clusterloader2/pkg/measurement/common/ooms_tracker.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/informer"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	clusterOOMsTrackerEnabledParamName = "clusterOOMsTrackerEnabled"
+	clusterOOMsTrackerName             = "ClusterOOMsTracker"
+	informerTimeout                    = time.Minute
+	oomEventReason                     = "OOMKilling"
+)
+
+var (
+	oomEventMsgRegex = regexp.MustCompile(`Kill process (\d+) \((.+)\) score \d+ or sacrifice child\nKilled process \d+ .+ total-vm:(\d+kB), anon-rss:\d+kB, file-rss:\d+kB.*`)
+)
+
+func init() {
+	if err := measurement.Register(clusterOOMsTrackerName, createClusterOOMsTrackerMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", clusterOOMsTrackerName, err)
+	}
+}
+
+func createClusterOOMsTrackerMeasurement() measurement.Measurement {
+	return &clusterOOMsTrackerMeasurement{}
+}
+
+type clusterOOMsTrackerMeasurement struct {
+	selector  *measurementutil.ObjectSelector
+	msgRegex  *regexp.Regexp
+	isRunning bool
+	stopCh    chan struct{}
+	oomsLock  sync.Mutex
+	ooms      []oomEvent
+}
+
+// Reevaluate if we can add new fields here when node-problem-detector starts
```

Comment:
nit: TODO: Reevaluate...
[to make it explicit that it's an AI]

Reply:
Done.